rust/src/domains/polars/expr/mod.rs

```diff
@@ -54,11 +56,14 @@ impl ExprContext {
                         .map(col)
                         .collect::<Vec<_>>(),
                 )
-                .agg([expr]),
+                .agg([expr.clone()]),
```

Comment:
I don't think `expr` is particularly heavyweight, but is there an argument for doing it once up top? Probably depends on context? Mostly for my own learning.

Reply:
The overhead is ultimately the same, as only one branch of the match will ever be taken at a time.